backend/src/schedule/schedule.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Schedule } from '.prisma/client';
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../global/prisma/prisma.service';
+import { AvailabilityEnum, ScheduleDTO } from './dto/schedule.dto';
+
+@Injectable()
+export class ScheduleService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  public async createSchedule(scheduleDTO: ScheduleDTO): Promise<ScheduleDTO> {
+    const createdSchedule = await this.prismaService.schedule.create({
+      data: scheduleDTO,
+    });
+
+    return ScheduleService.convertToDTO(createdSchedule);
+  }
+
+  public async getSchedules(): Promise<ScheduleDTO[]> {
+    const schedules = await this.prismaService.schedule.findMany();
+
+    return schedules.map((schedule) => ScheduleService.convertToDTO(schedule));
+  }
+
+  public async getScheduleById(id: string): Promise<ScheduleDTO> {
+    const schedule = await this.prismaService.schedule.findUniqueOrThrow({
+      where: { id },
+    });
+
+    return ScheduleService.convertToDTO(schedule);
+  }
+
+  public async editSchedule(
+    id: string,
+    scheduleDTO: ScheduleDTO
+  ): Promise<ScheduleDTO> {
+    const updatedSchedule = await this.prismaService.schedule.update({
+      data: scheduleDTO,
+      where: { id },
+    });
+
+    return ScheduleService.convertToDTO(updatedSchedule);
+  }
+
+  public async deleteSchedule(id: string): Promise<ScheduleDTO> {
+    const deleteSchedule = await this.prismaService.schedule.delete({
+      where: { id },
+    });
+
+    return ScheduleService.convertToDTO(deleteSchedule);
+  }
+
+  public static convertToDTO(schedule: Schedule): ScheduleDTO {
```

Comment:
@princhcanal ask lang ko princh kay curious ko, ngano naay convertToDTO ari nga method?

Reply:
kay ang return type sa Prisma lahi man usahay sa DTO. for example, ang `UserDTO` kay walay `password` field kay dili nato ganahan apilon ang password inig send sa backend sa mga users. pero ang `User` type sa Prisma kay naay password.

sa schedule kay same raman ila types pero always ko mag himo ani para consistent and in case in the future atong e change ang DTOs

Reply:
ang DTOs kay mao dapat ang e send/receive between frontend and backend

Reply:
I see, i see. Thank you princhg0d